amd_openvx_extensions/amd_nn/src/slice_layer.cpp

```diff
@@ -1,345 +1,386 @@
-            "void %s(__global float * in, uint in_offset, uint4 in_stride" // opencl_kernel_function_name
-            , (int)opencl_local_work[0], opencl_kernel_function_name);
-    }else
-    {
-        sprintf(item,
-            "#pragma OPENCL EXTENSION cl_khr_fp16 : enable\n"
-            "__kernel __attribute__((reqd_work_group_size(%d, 1, 1)))\n"    // opencl_local_work[0]
-            "void %s(__global half * in, uint in_offset, uint4 in_stride" // opencl_kernel_function_name
-            , (int)opencl_local_work[0], opencl_kernel_function_name);
+
+    for(int i=0; i<output_num_dims; i++) {
+        ocount *= output_dims[i];
     }
-    opencl_kernel_code = item;
-
-    for(int i = 0; i < num_outputs; i++) {
-        if (type == VX_TYPE_FLOAT32){
-        sprintf(item,
-            ",\n"
-            "                  __global float * out%d, uint out%d_offset, uint4 out%d_stride"  // i, i, i
-            , i, i, i);
-        }else
-        {
-            sprintf(item,
-                ",\n"
-                "                  __global half * out%d, uint out%d_offset, uint4 out%d_stride"  // i, i, i
-                , i, i, i);
+
+    // check the number of param dimensions and type
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_DIMS, &param_dims, sizeof(param_dims)));   
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_DATA_TYPE, &type, sizeof(type)));
+
+    // create array of input/indices/output tensors
+    vx_tensor input_tensor[param_dims], indices_tensor[param_dims], output_tensor[param_dims];
+    
+    // create context & graph
+    vx_context context = vxCreateContext();
+    status = vxGetStatus((vx_reference)context);
+    if(status) {
+        std::cerr << "ERROR: vxCreateContext() failed (" << status << ")" << std::endl;
+        return -1;
+    }
+
+    vxLoadKernels(context, "vx_nn"); // load vx_nn kernels
+
+    vx_graph graph = vxCreateGraph(context);
+    status = vxGetStatus((vx_reference)graph);
+    if(status) {
+        std::cerr << "ERROR: vxCreateGraph() failed (" << status << ")" << std::endl;
+        return -1;
+    }
+
+    // copy input tensor
+    status = vxMapTensorPatch((vx_tensor)parameters[0], 2, nullptr, nullptr, &map_id, stride, (void **)&fptr, VX_READ_ONLY, VX_MEMORY_TYPE_HOST, 0);
+    if(status) {
+        std::cerr << "ERROR: vxMapTensorPatch() failed for input tensor (" << status << ")" << std::endl;
+        return -1;
+    }
+    for(int i=0; i<8; i++) {
+        input.push_back((float)fptr[i]);
+    }
+    status = vxUnmapTensorPatch((vx_tensor)parameters[0], map_id);
+
+    // create tmp_input and copy input tensor
+    vx_tensor tmpi_tensor = vxCreateTensor(context, input_num_dims, input_dims, input_type, 0);
+    status = vxMapTensorPatch(tmpi_tensor, input_num_dims, nullptr, nullptr, &map_id, stride, (void **)&fptr, VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST, 0);
```

Comment:
Codacy found an issue: [Variable 'status' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5932170)